src/Node/Internal.hs

```diff
@@ -1161,12 +1169,7 @@ nodeDispatcher node handlerIn handlerInOut =
 
                             Channel.writeChannel channel Nothing
                             return channels'
-                        (Nothing, channels') -> do
-                            logWarning $ sformat "inconsistent peer and connection identifier state"
-                            return channels'
-                        (Just cState, _channels') -> do
-                            logError $ sformat ("unexpected ConnectionState in connectionLost: " % shown) cState
-                            throw $ InternalError "nodeDispatcher: connectionLost: impossible"
```

Comment:
For the record I have no idea why I ever wrote this. It's not impossible to lose a heavyweight connection while some lightweight connection is doing the handshake.